libpcp requires symbols which are currently available in static
libpgcommon library, link it here

--- pgpool-II-3.4.2/src/libs/pcp/Makefile.am~	2015-04-08 09:58:44.000000000 +0200
+++ pgpool-II-3.4.2/src/libs/pcp/Makefile.am	2015-06-18 14:08:24.100554508 +0200
@@ -1,6 +1,7 @@
 AM_CPPFLAGS = -D_GNU_SOURCE -DPOOL_PRIVATE -I @PGSQL_INCLUDE_DIR@
 
 lib_LTLIBRARIES = libpcp.la
+libpcp_la_LIBADD = $(PGSQL_LIB_DIR)/libpgcommon.a 
 libpcp_la_SOURCES = pcp.c \
 		    pcp_stream.c \
                     pcp_error.c \
